core/types/calls.cc

```diff
@@ -695,10 +695,6 @@ DispatchResult dispatchCallSymbol(const GlobalState &gs, const DispatchArgs &arg
         if (spec.flags.isKeyword) {
             break;
         }
-        if (ait + 1 == aend && hasKwargs && (spec.flags.isDefault || spec.flags.isRepeated) &&
-            Types::approximate(gs, arg->type, *constr).derivesFrom(gs, Symbols::Hash())) {
-            break;
-        }
```

Comment:
Does this logic need to be gated on the `--ruby3-keyword` flag or whatever we called it?

Reply:
No, It's just plain redundant now.